ext/storage_sources/s3_store/s3_connection.h

```diff
@@ -26,6 +26,7 @@ class S3Connection {
     int PutObject(const std::string &objectKey, const std::string &fileName) const;
     int DeleteObject(const std::string &objectKey) const;
     int ObjectExists(const std::string &objectKey, bool &exists, size_t &objectSize) const;
+    int BucketExists(bool &exists) const;
```

Comment:
Move this to the `private` section or add the bucket name argument depending on the solution you are going for given the other comment.

Reply:
done.